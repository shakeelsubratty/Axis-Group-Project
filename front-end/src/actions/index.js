import axios from 'axios';
import _ from 'lodash';


export const JOIN_WORKSHOP = 'join_workshop';
export const FETCH_IDEAS = 'fetch_ideas';
export const FETCH_ALL_IDEAS = 'fetch_all_ideas';
export const CREATE_IDEA = 'create_idea';
export const GET_WS_INFO = 'get_workshop_info';
export const CREATE_WORKSHOP = 'create_workshop';
export const SET_WORKSHOP_TO = 'set_workshop_to';
export const FETCH_USERS = 'fetch_users';
export const ATTEMPT_LOGIN = 'attempt_login';
export const LOG_OUT = 'log_out';
export const DELETE_IDEA = 'delete_idea';

let currentDataIdeas = [
	{ id: '0', title: "Elon Musk is a genious", explanation:'he has multiple successfull companies'},
	{ id: '1', title: "I should buy a Model S", explanation:'It is quick'},
	{ id: '2', title: "I should buy a Model X", explanation:'It is quick'},
];

let currentAllIdeas = [
	{ id: '0', title: "Elon Musk is a genious", explanation:'he has multiple successfull companies'},
	{ id: '1', title: "I should buy a Model S", explanation:'It is quick'},
	{ id: '2', title: "I should buy a Model X", explanation:'It is quick'},
	{ id: '3', title: "They are taking the hobbits to Isengard", explanation:'Saruman the White lives there'},
	{ id: '4', title: "I find your lack of faith disturbing.", explanation:'Dont be too proud of this technological terror you have constructed. The ability to destroy a planet is insignificant next to the power of the Force.'},
];

let currentDataUsers = [
	{ id: 1234567 },
	{ id: 4235433 },
	{ id: 3246346 },
	{ id: 5546473 },
	{ id: 4234254 },
	{ id: 4968573 },
	{ id: 4312142 },
	{ id: 4312143 },
	{ id: 9423342 },
	{ id: 4325463 },
	{ id: 9596734 },
	{ id: 6569234 },
	{ id: 4353452 },
	{ id: 1234552 },
	{ id: 6745634 },
	{ id: 1343242 },
];

export function attemptLogIn(username, password) {
  const request = true;
	console.log('login action called, username is ==> '+ username + ' password ==>', password);
	// we should make the post API call here.
  sessionStorage.setItem('usrn', username);
  sessionStorage.setItem('pass', password);
	return {
		type: ATTEMPT_LOGIN,
		payload: request
	};
}

export function logOut() {
  console.log('LOGGING OUT');
  sessionStorage.setItem('usrn', null);
  sessionStorage.setItem('pass', null);
  sessionStorage.setItem('wsId', null);
  return {
    type: LOG_OUT,
    payload: false
  };
}

// TODO: Change from boolean into userId
export function joinWorkshop(workshopId) {
	const ret = axios.get(`http://localhost:3000/participant/create/${workshopID}`);
	console.log('joinWorkshop Api -->', ret);
	const request = true;
	console.log('joinWorkshop action called, id is ==> ',workshopId);
	// we should make the post API call here.

	return {
		type: JOIN_WORKSHOP,
		payload: request
	};
}

export function createWorkshop(values) {
	// let result = null;
	// const ret =
	// 	console.log(response.data);
	// 	result = data.
  // });
	//
	// const request = '0123456789';
	// console.log('createWorkshop action called, values is ==> ',values);
	//
  // sessionStorage.setItem('wsId', result);
	// console.log('createWorkshop Api -->',result);
	//
	// return {
	// 	type: CREATE_WORKSHOP,
	// 	payload: result
	// };

	return (dispatch) => {

    axios.get(`http://localhost:3000/workshop/create?title=${values.title}&description=${values.description}`).then(function (response) {

			console.log('createWorkshop Api -->',response.data);
			sessionStorage.setItem('wsId', response.data);

      dispatch({
				type: CREATE_WORKSHOP,
				payload: response.data
      });
    }).catch((e) => {
      console.log(e);
    })
  }
}

export function setWorkshopTo(wsId){
  return {
    type: SET_WORKSHOP_TO,
    payload: wsId,
  };
}

// Api ready
export function fetchUsers(wsId){
	const ret = axios.get(`http://localhost:3000/workshop/view/${wsId}/users`);
	console.log('fetchUsers Api -->', ret);
  const request = currentDataUsers;
  console.log('fetchUsers returns==>', request);

  return {
    type: FETCH_USERS,
    payload: request,
  }
}

export function fetchIdeas(userId) {
	const ret = axios.get(`http://localhost:3000/participant/view/${userId}/ideas`);
	console.log('fetchIdeas Api -->', ret);

	let request = currentDataIdeas;
	console.log('fetchIdeas is called request ==>',request);

	return {
		type: FETCH_IDEAS,
		payload: request
	}
}

export function fetchAllIdeas(wsId){
	const ret = axios.get(`http://localhost:3000/workshop/view/${wsId}/ideas`);
	console.log('fetchAllIdeas Api -->', ret);

	let request = currentAllIdeas;
	console.log('fetchAllIdeas is called request ==>',request);

	return {
		type: FETCH_ALL_IDEAS,
		payload: request
	}
}

// TODO: Add userId and wsId as a value passed return idea id
export function createIdea(values, callback) {
	console.log('we are sending to the api==>',values)
		// here we send it to the API.
	//	const request = axios.post().then(()=>callback);
		const request = 'elonnnnnn'
		currentData.push(values);

		console.log(currentData);
		callback();
		return {
			type: CREATE_IDEA,
			payload: request
		}
}

export function getWorkshopInfo(wsId) {

	const ret = axios.get(`http://localhost:3000/workshop/view/${wsId}`);
	console.log('getWorkshopInfo Api -->', ret);

	const request = { title: 'May the force be with you', description: 'Did you ever hear the tragedy of Darth Plagueis The Wise? I thought not. It’s not a story the Jedi would tell you. It’s a Sith legend. Darth Plagueis was a Dark Lord of the Sith, so powerful and so wise he could use the Force to influence the midichlorians to create life…'}
	return {
		type: GET_WS_INFO,
		payload: request
	}
}

// TODO: id of idea and id of user
export function deleteIdea(id) {

	const request = 'aaaa'
	console.log('id passed ', id);
	// make api call to delete that idea.
	currentData.splice(id,1);

	console.log('delete idea called, result array=>',currentData);

	return{
		type: DELETE_IDEA,
		payload: request,
	}
}
