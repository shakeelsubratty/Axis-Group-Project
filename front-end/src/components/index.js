// All exports and imports for components.
import LoginFailed from './loginFailed';
import UserIdea from './userIdea';
import WorkshopIdea from './workshopIdea';
import LoadingScreen from './loadingScreen';
import WordCloud from './wordCloud';
import WorkshopClosed from './workshopClosed'

export { LoginFailed, UserIdea, WorkshopIdea, LoadingScreen, WordCloud, WorkshopClosed};
