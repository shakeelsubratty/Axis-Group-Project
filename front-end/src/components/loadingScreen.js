import React, { Component } from 'react';

/* Loading screen for users who have joined a workshop which is still not active.
Also used in case of connectivity issues.
*/
class LoadingScreen extends Component {
	constructor(props) {
		super(props)
	}

	// Render method for the component. Renders loading message and ws info if available.
	render(){
		if ( this.props.wsTitle != '' && this.props.wsDes != '') {
			return (
				<div className='main'>
					<div className='wrapper' style={{alignItems:'center', padding:'2%'}}>
						<h1 style={{color:'white'}}>{this.props.wsTitle}</h1>
						<h5 style={{color:'white', padding:'5px'}}>{this.props.wsDes}</h5>
						<div className='loader'/>
						</div>
				</div>
			);
		} else {
			return(
				<div className='main'>
					<div className='wrapper' style={{alignItems:'center', padding:'2%'}}>
						<h1 style={{color:'white'}}>Loading...</h1>
						<div className='loader'/>
						</div>
				</div>
			);
		}
  	}
}
export default LoadingScreen;
