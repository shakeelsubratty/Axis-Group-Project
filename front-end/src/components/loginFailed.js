import React, { Component } from 'react';
import { Link } from 'react-router-dom';

export default class LoginFailed extends Component {

  render(){
    return(
      <div className='main'>
        <div className='wrapper'>
          <div className='card card-big'>
            <h2 style={{textAlign:'center'}}>Oops...It looks like you are not logged in...</h2>
            <div className='card-body' style={{width:'100%', textAlign:'center'}}>
              <Link className='btn btn-primary' style={{margin:'5%', width:'50%'}} to='/login'>
                Log in now
              </Link>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
