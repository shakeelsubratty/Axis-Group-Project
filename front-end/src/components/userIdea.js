import React, {Component} from 'react';
import PropTypes from 'prop-types';

// Component which displays a user idea in a collapsible container for a participant
export default class UserIdea extends Component {

    constructor(props){
        super(props)
        this.state = {
            isExpanded: false
        }
    }

	// Function called when card is clicked.
   click(event){
      event.preventDefault();
      this.setState({
         isExpanded: !this.state.isExpanded,
         height: this.refs.inner.clientHeight
      })
   }

	// Function called when delete button is pressed.
	deleteClick() {
		console.log('onclick called');
		console.log(this.props.id);
		this.props.callback(this.props.id);
	}

	// Render method for the component. Renders ideas and form to submit them.
   render() {
   	const {title, children, id} = this.props;
      const {isExpanded, height} = this.state;
      const currentHeight = isExpanded ? height : 0;
      return (
      	<div className={`card panel ${isExpanded ? 'is-expanded' : ''}`} >
         	<div className="card-header panel-heading" style={{display: 'flex'}} onClick={(event) => this.click(event)} >
					<h4>{title}</h4>
            </div>
            <div className="panel-collapse" style={{height: currentHeight+'px'}}>
               <div className="card-body panel-body" ref="inner" style={{padding: '3%'}}>
                  {children}
						<div className='button-box' style={{display: 'flex', justifyContent:'flex-end'}}>
							<button type='button' className='btn btn-danger' onClick={this.deleteClick.bind(this)} >Delete</button>
						</div>
               </div>

            </div>
      	</div>
      )
   }
}

UserIdea.propTypes = {
   title: PropTypes.string,
};
