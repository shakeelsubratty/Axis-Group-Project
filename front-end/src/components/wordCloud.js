import React, { Component } from 'react';
import { TagCloud } from 'react-tagcloud';

// const data = [
//   { value: "JavaScript", count: 38, color:'#000'}, //by adding color you can make it a specific color
//   { value: "React", count: 30 },
//   { value: "Nodejs", count: 28 },
//   { value: "Express.js", count: 25 },
//   { value: "HTML5", count: 33 },
//   { value: "MongoDB", count: 18 },
//   { value: "CSS3", count: 20 }
// ];

// Component that displays WordCloud
class WordCloud extends Component {
  constructor(props){
    super(props);
  }

  	// Render method for the component. Renders ideas and form to submit them.
	render(){
		return(
      	<TagCloud minSize={12}
            maxSize={35}
            tags={this.props.data}
            shuffle={false}
            onClick={tag => alert(`'${tag.value}' was selected!`)} />
		);
  	}
}

export default WordCloud;
