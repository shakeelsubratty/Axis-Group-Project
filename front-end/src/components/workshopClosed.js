import React, { Component } from 'react';
import { Link } from 'react-router-dom';

export class WorkshopClosed extends Component {
	constructor(props) {
		super(props)
	}

  render(){

		if ( this.props.wsTitle != '') {

			return (
				<div className='main'>
					<div className='wrapper' style={{alignItems:'center', padding:'2%'}}>
		       <h1 style={{color:'white'}}>{this.props.wsTitle}</h1>
            <h5 style={{color:'white', padding:'5px'}}>This workshop has been closed</h5>
            <Link className='btn btn-danger' to='/enter-workshop'>
              Exit
            </Link>
          </div>
				</div>
			);
		} else {
			return(
        <div className='main'>
					<div className='wrapper' style={{alignItems:'center', padding:'2%'}}>
            <h5 style={{color:'white', padding:'5px'}}>This workshop has been closed</h5>
            <Link className='btn btn-danger' to='/enter-workshop'>
              Exit
            </Link>
          </div>
				</div>
			);
		}
  }
}
export default WorkshopClosed;
