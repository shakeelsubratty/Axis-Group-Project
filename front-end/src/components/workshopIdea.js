import React, { Component } from 'react';
//import PropTypes from 'prop-types';

class WorkshopIdea extends Component {

    constructor(props){
        super(props)
        this.state = {
            isExpanded: false
        }
    }

    toggle(event){
        event.preventDefault();
        this.setState({
            isExpanded: !this.state.isExpanded,
            height: this.refs.inner.clientHeight
        })
    }

    render(){
        const {title, children, id} = this.props;
        const {isExpanded, height} = this.state;
        const currentHeight = isExpanded ? height : 0;
        return (
            <div className={`card panel ${isExpanded ? 'is-expanded' : ''}`} >
                <div className="card-header panel-heading" style={{display: 'flex'}} onClick={(event) => this.toggle(event)} >
						      <h4>{title}</h4>
                </div>
                <div className="panel-collapse" style={{height: currentHeight+'px'}}>
                    <div className="card-body panel-body" ref="inner" style={{padding: '3%'}}>
                        {children}
                    </div>

                </div>
            </div>
        )
    }

}

// WorkshopIdea.propTypes = {
//     title: PropTypes.string,
// };

export default WorkshopIdea;
