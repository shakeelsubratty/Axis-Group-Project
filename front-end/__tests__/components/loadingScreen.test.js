import React from 'react';
import { LoadingScreen } from '../../src/components';
import renderer from 'react-test-renderer';


describe("LoadingScreen", () => {
	it('matches snapshot', () => {
		const tree = renderer.create(<LoadingScreen />).toJSON();
		expect(tree).toMatchSnapshot();

	});
});
