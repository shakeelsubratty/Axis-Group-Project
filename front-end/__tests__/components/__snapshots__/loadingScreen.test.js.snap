// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingScreen matches snapshot 1`] = `
<div
  className="main"
>
  <div
    className="wrapper"
    style={
      Object {
        "alignItems": "center",
        "padding": "2%",
      }
    }
  >
    <h1
      style={
        Object {
          "color": "white",
        }
      }
    />
    <h5
      style={
        Object {
          "color": "white",
          "padding": "5px",
        }
      }
    />
    <div
      className="loader"
    />
  </div>
</div>
`;
